velox/docs/functions/presto/conversion.rst

```diff
@@ -919,6 +979,54 @@ Invalid example
   SELECT cast('-3E+2.1' as decimal(12, 2)); -- Value is not a number
   SELECT cast('3E+' as decimal(12, 2)); -- Value is not a number
 
+Cast to IPAddress
+---------------
+
+From varchar type
+^^^^^^^^^^^^^^^^^
+
+To cast a varchar to IPAddress it must be in the form of either
+IPV4 or IPV6. It can be written in full form or shortened form.
+All IPV6 addresses that have the form of an IPV4 mapped IPV6
+address will be assumed to be IPV4 addresses. 
+
+Valid examples
+
+::
+
+  SELECT cast('2001:0db8:0000:0000:0000:ff00:0042:8329' as ipaddress); -- ipaddress '2001:db8::ff00:42:8329'
+  SELECT cast('1.2.3.4' as ipaddress); -- ipaddress '1.2.3.4'
+  SELECT cast('::ffff:ffff:ffff' as ipaddress); -- ipaddress '255.255.255.255'
+
+Invalid examples
+
+::
+
+  SELECT cast('2001:db8::1::1' as ipaddress); -- Invalid IP address
+  SELECT cast('789.1.1.1' as ipaddress); -- Invalid IP address
+
+From varbinary type
+^^^^^^^^^^^^^^^^^
+
+To cast a varbinary to IPAddress it must be either IPV4(4 Bytes)
+or IPV6(16 Bytes) in network byte order.
+All IPV6 addresses that have the form of an IPV4 mapped IPV6
```

Comment:
IPv6 addresses in the form of an IPv4 mapped address are assumed ...

BTW, what does this mean exactly? Would you clarify?